mlir/lib/ExecutionEngine/conv-validation-wrappers.cpp

```diff
@@ -558,9 +558,41 @@ extern "C" void mcpuVerifyInt32(int32_t *gpuAllocated, int32_t *gpuAligned,
                                 int64_t gpuStride, int32_t *valAllocated,
                                 int32_t *valAligned, int64_t valOffset,
                                 int64_t valSize, int64_t valStride,
-                                float thr_RMS, float thr_absDiff,
-                                float thr_relDiff, char printDebug) {
+                                char printDebug) {
+  assert(gpuSize == valSize);
+  for (int64_t i = 0; i < valSize; ++i) {
+    if (gpuAligned[i] != valAligned[i]) {
+      if (printDebug)
+        printf("gpu=%d val=%d\n", gpuAligned[i], valAligned[i]);
+      printf("[%d %d %d]\n", 0, 0, 0);
```

Comment:
OK. I'll make it just print -1.

Reply:
@zhanglx13 I disagree. We can think of integers as still using the metrics, except all the metric values are set to `0`.

Reply:
So it should still be `[0, 0, 0]` here

Reply:
Yeah, that's a good interpretation.
I was thinking [0,0,0] means the difference is larger than the threshold. So in the float case, we usually go and check to see if we need to set a larger threshold or there is really a bug. For integer case, [0,0,0] means there is a bug. If we keep using [0,0,0], we need to keep in mind that they mean different things for float and integers.

Reply:
They ... don't? At least not how I'm thinking of it. An integer computation is one where we don't have acceptable rounding losses at all, so the max RMS error = the max absolute difference = the max relative difference = 0.

Reply:
I was saying [0,0,0] has different meanings for int and float.
For int, it means the test fails because of a bug.
For float, it does not always indicate a bug. It usually requires further investigation to identify the issue.
Such information is desired when we see the results from lit. Either it is embedded in the verification result ([-1,-1,-1] means a bug for the int test) or we need to check the data type manually for the test (which is not in the output of lit since the test does not have datatype in their filenames).
But I don't think this makes a big difference since we will examine the failed tests anyway. So I'm ok with either method.

Reply:
Ah.

I'd want to stick with [0, 0, 0] for consistency.

And I'd at least argue that float tests that aren't `[1, 1, 1]` at least deserve to be looked at very carefully.

Reply:
Sure. @yiqian1 you can keep it unless you have a different opinion on this.

Reply:
No, I don't have a strong opinion on this.